File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxi = 0;
        int CurrCount = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i] == 1){
                CurrCount++;
            }
            else{
                CurrCount = 0;
            }
            maxi = Math.max(maxi,CurrCount);
        }
        return maxi;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int i = 0;
        int j = 0;
        int count = 0;
        int ans=0;
        // while (i < j) {
        //     if (nums[i] == nums[j]) {
        //         count++;
        //         j++;
        //     }
        //      if (nums[i] != nums[j]) {
        //         count=1;
        //         i++;
        //     }
        // }

        // return count;
        for(;j<nums.size();j++){
            if(nums[j]==0){
                i=j+1;
            }
            ans = max(ans,j-i+1);
        }
        return ans;
    }
};